src/pages/ProductDetail/index.js

```diff
@@ -1,46 +1,76 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import { loadProduct } from '../../store/product/actions'
+import { addProduct } from '../../store/cartItems/actions'
 
-import { getAuthRequest } from '../../api/getAuthRequest'
 import Layout from '../../components/Layout'
 import Loader from '../../components/Loader'
 import { H1 } from '../../components/Typography'
 import ProductDetailComponent from './components/ProductDetail'
+
+import { getAuthRequest } from '../../api/getAuthRequest'
 import { productWithPrice } from '../../helpers/transform/productWithPrice'
+import { AddButton } from '../ProductList/components/Product/styled'
 
-class ProductDetail extends React.Component {
+class Product extends React.Component {
   state = {
     isLoading: true, // stop confusing users, what happening until fetch data
-    product: {},
   }
 
   async componentDidMount() {
-    const { match } = this.props
+    if (this.props.product.length === 0) {
```

Comment:
there should be null check after default state of product state will be set to null

Reply:
Fixed!